s3fs/core.py

```diff
@@ -644,7 +763,7 @@ def isdir(self, path):
             return False
 
         # This only returns things within the path and NOT the path object itself
-        return bool(self._lsdir(path))
+        return bool(sync(self.loop, self._lsdir, path))
 
     def ls(self, path, detail=False, refresh=False, **kwargs):
```

Comment:
I'm curious, so for something like `ls` is there an async equivalent?

Reply:
For the filepath management, I only plumbed in as much as seemed required to make the bulk operations work. So _info is async (calls head_object, or gets from cache) because copy needs to know how big the file is. I want to try to steer clear of writing async versions of all of the default implementations in AbstractFileSystem for now.